rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -482,71 +523,61 @@ func (r *Layer2Relayer) ProcessCommittedBatches() {
 		)
 		if err != nil {
 			log.Error("Pack finalizeBatchWithProof failed", "err", err)
-			return
+			return err
 		}
-
-		txID := hash + "-finalize"
-		// add suffix `-finalize` to avoid duplication with commit tx in unit tests
-		txHash, err := r.finalizeSender.SendTransaction(txID, &r.cfg.RollupContractAddress, big.NewInt(0), data, 0)
-		finalizeTxHash := &txHash
+	} else {
+		var err error
+		txCalldata, err = r.l1RollupABI.Pack(
+			"finalizeBatch",
```

Comment:
You need to update abi to use `finalizeBatch`.

Reply:
Yes. Updated in [78af5a9](https://github.com/scroll-tech/scroll/pull/1004/commits/78af5a9dbbb9e043318cef4454258293aae4ff7f)
also added a unit test.